config/exported_config.go

```diff
@@ -3,7 +3,12 @@ package config
 // ExportedConfig is a struct that outlines a set of configuration.
 // Each module is supposed to emit ExportedConfig into DI, and Package config should collect them.
 type ExportedConfig struct {
-	Owner   string
-	Data    map[string]interface{}
-	Comment string
+	Owner    string
+	Data     map[string]interface{}
+	Comment  string
+	Validate Validator
 }
+
+// Validator is a method to verify if config is valid. If it is not valid, the
+// returned error should contain a human readable description of why.
+type Validator func(data map[string]interface{}) error
```

Comment:
Why not use type `ConfigAccessor` as argument？

Reply:
ConfigAccessor is an interface. Data is data. Validation validates the data, not the interface.

Reply:
Interfaces describes behavior, not data.